internal/services/automation/automation_variables_data_source.go

```diff
@@ -0,0 +1,289 @@
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+
+						"encrypted": {
+							Type:     pluginsdk.TypeBool,
+							Computed: true,
+						},
+
+						"value": {
+							Type:     pluginsdk.TypeInt,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"null": {
+				Type:     pluginsdk.TypeList,
+				Computed: true,
+				Elem: &pluginsdk.Resource{
+					// How do I make this related to the specific type?
+					Schema: map[string]*pluginsdk.Schema{
+						"name": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+
+						"description": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+
+						"encrypted": {
+							Type:     pluginsdk.TypeBool,
+							Computed: true,
+						},
+
+						"value": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"string": {
+				Type:     pluginsdk.TypeList,
+				Computed: true,
+				Elem: &pluginsdk.Resource{
+					// How do I make this related to the specific type?
+					Schema: map[string]*pluginsdk.Schema{
+						"name": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+
+						"description": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+
+						"encrypted": {
+							Type:     pluginsdk.TypeBool,
+							Computed: true,
+						},
+
+						"value": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceAutomationVariablesRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Automation.VariableClient
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	automationId := variable.NewAutomationAccountID(subscriptionId, d.Get("resource_group_name").(string), d.Get("automation_account_name").(string))
+
+	variableList, err := client.ListByAutomationAccountComplete(ctx, automationId)
+	if err != nil {
+		return fmt.Errorf("listing variables in Automation Account %q: %+v", automationId, err)
+	}
+
+	d.SetId(automationId.ID())
+	d.Set("resource_group_name", automationId.ResourceGroupName)
+	d.Set("automation_account_name", automationId.AutomationAccountName)
+
+	var var_bool []map[string]interface{}
+	var var_dt []map[string]interface{}
+	var var_encrypt []map[string]interface{}
+	var var_int []map[string]interface{}
+	var var_null []map[string]interface{}
+	var var_str []map[string]interface{}
+
+	for _, v := range variableList.Items {
+		_, err := variable.ParseVariableID(*v.Id)
+		if err != nil {
+			return err
+		}
+
+		res := map[string]interface{}{
+			"name":        *v.Name,
+			"description": *v.Properties.Description,
+			"encrypted":   *v.Properties.IsEncrypted,
```

Comment:
```suggestion
			"name":        pointer.From(v.Name),
			"description": pointer.From(v.Properties.Description),
			"encrypted":   pointer.From(v.Properties.IsEncrypted),
```

Reply:
Should this apply to _all_ pointer references in the code?